packages/platform-sdk/test/integration-tests/CleanEverythingIntegrationTest.ts

```diff
@@ -0,0 +1,242 @@
+  try {
+    await apiRoot[resourceType]()
+      .withId({ ID: resourceId })
+      .delete({ queryArgs: { version } })
+      .execute()
+    console.log(`Deleted ${resourceType.slice(0, -1)} with ID: ${resourceId}`)
+  } catch (error) {
+    console.error(
+      `Error deleting ${resourceType.slice(0, -1)} with ID: ${resourceId}:`,
+      error
+    )
+  }
+}
+
+// Function to unpublish and delete a product
+const unpublishAndDeleteProduct = async (product: Product) => {
+  if (product.masterData.published) {
+    console.log(`Unpublishing product: ${product.id}`)
+    await apiRoot
+      .products()
+      .withId({ ID: product.id })
+      .post({
+        body: {
+          version: product.version,
+          actions: [{ action: 'unpublish' }],
+        },
+      })
+      .execute()
+    console.log(`Product ${product.id} unpublished.`)
+  }
+
+  // Increment version after unpublish
+  await deleteResource(
+    'products',
+    product.id,
+    product.version + (product.masterData.published ? 1 : 0)
+  )
+}
+
+// Function to delete all products for a specific product type
+const deleteProductsForProductType = async (productTypeId: string) => {
+  const products = await apiRoot
+    .products()
+    .get({ queryArgs: { where: `productType(id="${productTypeId}")` } })
+    .execute()
+
+  console.log(
+    `Found ${products.body.results.length} products for product type ${productTypeId}...`
+  )
+  return Promise.all(
+    products.body.results.map((product: Product) =>
+      unpublishAndDeleteProduct(product)
+    )
+  )
+}
+
+// Function to delete all products in the system
+const deleteAllProducts = async () => {
+  const products = await apiRoot.products().get().execute()
+  console.log(`Found ${products.body.results.length} products to delete...`)
+
+  return Promise.all(
+    products.body.results.map(async (product: Product) => {
+      await unpublishAndDeleteProduct(product)
+    })
+  )
+}
+
+// Generic function to check if a resource is referenced by another resource and delete it if not
+const safeDeleteResource = async (
+  resourceType,
+  resourceId,
+  version,
+  referenceCheckFn
+) => {
+  const isReferenced = await referenceCheckFn()
+  if (isReferenced) {
+    console.log(
+      `${resourceType.slice(0, -1)} with ID: ${resourceId} is referenced. Skipping deletion.`
+    )
+    return
+  }
+  await deleteResource(resourceType, resourceId, version)
+}
+
+// Main function to delete all resources
+const deleteAllResources = async () => {
+  try {
+    // Product Types and associated products
+    const productTypes = await apiRoot.productTypes().get().execute()
+    await Promise.all(
+      productTypes.body.results.map(async (productType) => {
+        await deleteProductsForProductType(productType.id)
+      })
+    )
+    console.log(`Deleted ${productTypes.body.results.length} Product Types`)
+
+    // Delete all standalone products (not tied to a product type)
```

Comment:
Sorry, can you clarify this point - I thought products need always to be tied to a product type? https://docs.commercetools.com/api/projects/products#product

Reply:
@cneijenhuis Thanks for pointing that out! Yes, you're right, products are always tied to a product type. However, when deleting a product, we don't need to reference the product type explicitly in the deletion process. In this case, I'm fetching all the products and deleting them directly. The link to the product type is part of the product’s metadata, but it doesn’t need to be called out specifically for the deletion to happen. I focus on unpublishing and then deleting the product itself. 

Let me know if that makes sense!